ansible_runner/config/_base.py

```diff
@@ -117,11 +117,13 @@ def __init__(self,
             self.cwd = self.project_dir
         if self.cwd:
             self.cwd = os.path.abspath(self.cwd)
-
-        if not os.path.exists(self.private_data_dir):
-            os.makedirs(self.private_data_dir, mode=0o700)
-        if not os.path.exists(self.artifact_dir):
-            os.makedirs(self.artifact_dir, mode=0o700)
+        try:
+            if not os.path.exists(self.private_data_dir):
+                os.makedirs(self.private_data_dir, mode=0o700)
+            if not os.path.exists(self.artifact_dir):
+                os.makedirs(self.artifact_dir, mode=0o700)
+        except FileExistsError:
```

Comment:
This is just to deal with a race condition between the path.exists conditional and the makedirs conditional? I think I'd rather just use the parameter `exist_ok=True` on `os.makedirs` ... we're no longer support pre-3.0 python versions so we don't have to worry about the parameter not being available:

https://docs.python.org/3/library/os.html#os.makedirs

Reply:
Fixed